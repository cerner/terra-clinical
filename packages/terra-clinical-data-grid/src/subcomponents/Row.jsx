import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import memoize from 'memoize-one';

import styles from './Row.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * String identifier of the section in which the Row will be rendered.
   */
  sectionId: PropTypes.string.isRequired,
  /**
   * String identifier for the Row.
   */
  rowId: PropTypes.string.isRequired,
  /**
   * String-formatted width that the Row should be rendered as. Any valid css width value is supported (i.e. 200px, 3rem).
   */
  width: PropTypes.string.isRequired,
  /**
   * String-formatted height that the Row should be rendered as. Any valid css width value is supported (i.e. 200px, 3rem).
   */
  height: PropTypes.string.isRequired,
  /**
   * Boolean indicating whether or not the Row should render as selected.
   */
  isSelected: PropTypes.bool,
  /**
   * Content to render within the Row.
   */
  children: PropTypes.node,
};

class Row extends React.Component {
  constructor(props) {
    super(props);

    this.getRowStyles = memoize(this.getRowStyles);
  }

  /* eslint-disable class-methods-use-this */
  getRowStyles(width, height) {
    return {
      width,
      height,
    };
  }
  /* eslint-enable class-methods-use-this */

  render() {
    const {
      rowId,
      sectionId,
      width,
      height,
      isSelected,
      children,
      ...customProps
    } = this.props;

    return (
      <div
        {...customProps}
        className={cx(['row', { selected: isSelected }, customProps.className])}
        style={this.getRowStyles(width, height)}
        data-row
        data-row-id={rowId}
        data-section-id={sectionId}
      >
        {children}
      </div>
    );
  }
}

Row.propTypes = propTypes;

export default Row;
