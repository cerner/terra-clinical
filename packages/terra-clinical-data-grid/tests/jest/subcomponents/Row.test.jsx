import React from 'react';
import Row from '../../../src/subcomponents/Row';

describe('Row Snapshot Tests', () => {
  it('should render a Row with minimal props', () => {
    const row = shallow((
      <Row
        sectionId="section-test"
        rowId="row-test"
        width="123px"
        height="4rem"
      />
    ));

    expect(row).toMatchSnapshot();
  });

  it('should render a Row with all provided props', () => {
    const row = shallow((
      <Row
        sectionId="section-test"
        rowId="row-test"
        width="123px"
        height="4rem"
        isSelected
      >
        <div>Row Content</div>
      </Row>
    ));

    expect(row).toMatchSnapshot();
  });
});
