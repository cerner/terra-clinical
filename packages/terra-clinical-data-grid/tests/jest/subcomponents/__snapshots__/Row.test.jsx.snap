// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Snapshot Tests should render a Row with all provided props 1`] = `
<div
  className="row selected"
  data-row={true}
  data-row-id="row-test"
  data-section-id="section-test"
  style={
    Object {
      "height": "4rem",
      "width": "123px",
    }
  }
>
  <div>
    Row Content
  </div>
</div>
`;

exports[`Row Snapshot Tests should render a Row with minimal props 1`] = `
<div
  className="row"
  data-row={true}
  data-row-id="row-test"
  data-section-id="section-test"
  style={
    Object {
      "height": "4rem",
      "width": "123px",
    }
  }
/>
`;
